core/src/view/view.cpp

```diff
@@ -228,23 +243,23 @@ LngLat View::getCenterCoordinates() const {
     return center;
 }
 
-void View::update(bool _constrainToWorldBounds) {
+void View::applyConstraint() {
+    // Approximate the view diameter in pixels by taking the maximum dimension.
+    double viewDiameterPixels = std::fmax(getWidth(), getHeight()) / pixelScale();
+    // Approximate the minimum zoom that keeps with view span within the drawable projection area. [1]
+    double minZoom = std::log(viewDiameterPixels / MapProjection::tileSize() + 2) / std::log(2);
```

Comment:
I was trying to avoid calling these every time zoom or position is set and only call `applyConstraint` when either the size gets updated or pixelScale changes. (Yes I missed `setPosition` in my PR :| ).